### PR TITLE
fix: preventing redirects from live preview

### PR DESCRIPTION
**Pull request by VibhorJaiswal**
fixes: #6843 

Hi! as discussed in PR #6852  `to prevent the action on all anchors within code examples within the playground component`, here is the revised solution.

### CODE REVIEW

File: www/src/theme/Playground/Preview.tsx [hunk 1]
```diff
@@ -55,9 +55,17 @@ const Preview: React.FC<PreviewProps> = ({ className }) => {
     },
   );
 
+
+  const handleCustomRedirect = (e) => {
+    const target = e.target as HTMLElement;
+    if (target.closest('a[href]')) {
```
Comment (kyletsang) [inline www/src/theme/Playground/Preview.tsx:61]: Can you try this?

```
if (e.target.tagName === 'A') {
  e.preventDefault();
}
```

File: www/src/theme/Playground/Preview.tsx [hunk 2]
```diff
@@ -55,9 +55,17 @@ const Preview: React.FC<PreviewProps> = ({ className }) => {
     },
   );
 
+
+  const handleCustomRedirect = (e) => {
```
Comment (kyletsang) [inline www/src/theme/Playground/Preview.tsx:59]: Can you wrap this callback with a `useEventCallback` from `@restart/hooks`.  Also add a type to `e` and remove the cast on line 60

**Comment by VibhorJaiswal** [comment]
Hi! @kyletsang, I've implemented the changes based on your suggestions. Could you please review and let me know if it looks good now?

**Review by kyletsang — APPROVED** [review]

**Comment by kyletsang** [comment]
Thanks

**Pull request merged**